tern/classes/file_data.py

```diff
@@ -262,3 +262,15 @@ def merge(self, other):
                     self.origins.add_notice_to_origins(o.origin_str, n)
             return True
         return False
+
+    def is_equal(self, other):
+        '''This method performs a deep comparison between two objects by
+        iterating over the dictionary for each object returned by to_dict
+        and comparing the vales for each key in both. This function will only
+        return true if every key value pair matches between the two
+        dictionaries.'''
```

Comment:
I don't think we need to compare all the properties in the FileData object. Just the file name, path and checksum.